### ISSUE TITLE
ISDK-2536: Support for Catalyst/XCFrameworks

### ISSUE THREAD

**Issue by lilhinx** [twilio/twilio-video-ios#49]
TwilioVideo.framework cannot be linked to projects which target UIKit for Mac 13.0. Without this capability, TwilioVideo cannot be used on this exciting new platform.

**Comment by ceaglest** [comment]
Hi @lilhinx,

I am also excited for the future of UIKit on the Mac! In the past we have experimented with an AppKit based macOS SDK for Twilio Video, but it was a non-trivial amount of effort to account for the differences between AppKit and UIKit.

I will discuss with the team and share more info once I have it.

Best,
Chris

**Comment by lilhinx** [comment]
Hi, very interested in this. Are there any updates?

**Comment by ceaglest** [comment]
Hello,

Since the end goal is to support Catalyst, I see 3 major steps to getting there:

1. Migrate the Twilio Video CI pipeline to use Xcode 11.0 and the iOS 13.0 SDK.
2. Package and distribute the SDK as an XCFramework with currently supported iOS & simulator architectures.
3. Add a macOS / catalyst slice to the XCFramework.

I am currently working on the first step now that the Xcode 11 GM artifact is available. The second and third steps are something we will consider further once we are happy with the stability of Twilio Video on iOS 13 and iPadOS 13 (see the [Migration Guide](https://github.com/twilio/twilio-video-ios/blob/Releases/iOS-13-Migration-Guide.md) for a sense of what we are working on).

Best,
Chris

**Comment by cyrilzakka** [comment]
Wanted to voice my support for this as well! How's everything coming along?

**Comment by ceaglest** [comment]
Hi,

With the 3.0 SDK generally available, and built with Xcode 11 we have completed step 1. We have also resolved the majority of integration issues with iOS 13 and iPadOS 13.

There is interest in XCFramework packaging for a number of our SDKs (including voice). Getting this done would bring us a step closer to Catalyst support. Unfortunately, I don't have a timeline to share yet on when we will be able to complete steps 2 and 3. I will check back in January 2020 with another update once I have had time to discuss again with stakeholders.

Thank you for your feedback about Catalyst support.

Best,
Chris

**Comment by cyrilzakka** [comment]
Thank you so much for your work and for continuing to provide us with updates!

**Comment by Hamoonist** [comment]
I'm here to express my support and thank you for working on it!  <3

**Comment by koystinen** [comment]
Hoping the work on catalyst support is moving along and we can get an update soon, as we really really need this 😅

**Comment by lilhinx** [comment]
I think about this every day :)

**Comment by ArifADS** [comment]
I'm more in need for Twilio Chat Client, but i'm here to express my support too :)

**Comment by toseef** [comment]
Hi,
Is there any update of twillio support for macos?

**Comment by piyushtank** [comment]
Hi all,

We are not actively working on this feature right now, but we consider pulling it in. We will post an update here when we have more information.

Best,
Piyush

**Comment by echamussy** [comment]
Yes please!!! Catalyst support would be huge for us! Please keep us updated

**Comment by jkrzemie** [comment]
Given the latest WWDC announcements, is there any updated decision in terms of planning for this?

Best
Jacob

**Comment by ceaglest** [comment]
Hi,

I can't give any updates on if we will support Catalyst yet, but we are participating in sessions and gathering information at WWDC this week.

The subject of running iOS and iPadOS apps on a Mac with Apple Silicon probably deserves a different issue.

Best,
Chris

**Comment by swells808** [comment]
Its September now are there any updates?

**Comment by paynerc** [comment]
@swells808,

I can confirm that with the upcoming `4.0` release of TwilioVideo, we will be shipping as an `.xcframework`. There are still some things that need to be worked out before we can support Catalyst, but it is being investigated at this time.

Ryan

**Comment by OlivierGuy13** [comment]
I can't wait until this update !! 
Please keep us updated
Thank you for working on it !!! :D

**Comment by piyushtank** [comment]
We have just shipped 4.0.0-beta1 as  `.xcframework`. 
[Here](https://www.twilio.com/docs/video/video-ios-4x-beta) are the changelogs. Try it out and let us know if you have any feedbacks or questions.

**Comment by jigar007** [comment]
@piyushtank @paynerc @ceaglest Hi do you have any roughly estimated timeline on when 4.0.0 one production version will be available? so we can plan it better and decide which version of SDK we should use. Thanks.

**Comment by AkrutiPatel1311** [comment]
@ceaglest any update on Catalyst support for Twilio Voice????

**Comment by chetan1406** [comment]
@piyushtank @paynerc @ceaglest any update on Catalyst support for Twilio Voice????

**Comment by pulimento** [comment]
That would be very useful! How we can help?

**Comment by janremes** [comment]
@piyushtank Any update on this one? Thanks!

**Comment by paynerc** [comment]
Folks,

Twilio Video iOS `4.0.0` was released on December 4th and is packaged as a `.xcframework` which also allows us to deliver it as a Swift Package.

We are still investigating what it will take to fully support Catalyst.

Ryan

**Comment by gertjanjansen** [comment]
Hi Ryan,

It would be really great if you could support Catalyst. Do you have any idea when you will decide about supporting Catalyst or not?

Thanks,
Gertjan

**Comment by VictorPapyshev** [comment]
Hello Ryan,

If you are able to provide one, an update on the status of Catalyst support with Twilio Video would be much appreciated!

Best,
Victor

**Comment by ukudala** [comment]
Hi Ryan! I would like to add another vote of excitement and desire for Twilio Video Catalyst support. I can see this thread has been long active, and it's my hope that you guys are nearing a release. Thank you for bringing Twilio Video to Mac Catalyst ❤️

Best regards,
Jordan, Scorebreak

**Comment by mweyamutsvene** [comment]
Hi Ryan,
Just to piggy back on this thread, are there plans to also have Twilio Conversation Catalyst support as well?  

~tommy

**Comment by KrisConrad** [comment]
Any movement, or any kind of update at all for those of us that have been eagerly awaiting Catalyst support?

**Comment by anonyome-michael** [comment]
Hi, is there any update on Catalyst support for TwilioVideo?

**Comment by ukudala** [comment]
Hi Ryry, adding another vote for Catalyst support here!! UIKit ❤️❤️

**Comment by cvcoder1** [comment]
This would be a fantastic spec to finish. I know 20+ teams that would implement immediately. Any way to beta test or help support, let me know.

**Comment by KrisConrad** [comment]
Yeah, count me in on a beta if that happens. I've been hoping for Catalyst support for quite a while, my project and our users are very ready!

**Comment by ghost** [comment]
Any movement on this? It would be fantastic!

**Comment by ghost** [comment]
> Folks,
> 
> Twilio Video iOS `4.0.0` was released on December 4th and is packaged as a `.xcframework` which also allows us to deliver it as a Swift Package.
> 
> We are still investigating what it will take to fully support Catalyst.
> 
> Ryan

Hi Ryan, 

Could you please provide an update on this?

**Comment by KrisConrad** [comment]
Soooooooo 🦗🦗🦗?

**Comment by colincavil112** [comment]
@piyushtank - Hi, is there any update on this? It has been nearly three years since it was first mentioned, and clearly by the looks of the thread it has highly desired.. my team and I are waiting on this and now must seek (in the interim) TwilioVideo alternatives, unfortunately because it is the only third party library out of 30+ we use which is not compatible..

**Comment by ghost** [comment]
Has this really not been updated to support Catalyst yet? Yikes.. no wonder people are starting to use other frameworks, it's sad to see Twilio fall apart like this.

**Comment by ghost** [comment]
Hello. Is there any support for catalyst in the new version? My team is want to use Twilio but our new app to come out on the app store is using catalyst so we cannot use Twilio Programmable Video yet :/ @piyushtank @kierandonaldson

**Comment by ukudala** [comment]
@piyushtank @kierandonaldson bumping this ☝️

**Comment by YoavGro** [comment]
Hi any news on the matter?

**Comment by lilhinx** [comment]
@ceaglest I'd like to chime in roughly 3 years after I asked the initial question. 

1. I'd still love to see this happen.
2. Catalyst is no longer super important to me, but macOS is.

**Comment by jimmycooks9** [comment]
> @ceaglest I'd like to chime in roughly 3 years after I asked the initial question.
> 
> 1. I'd still love to see this happen.
> 2. Catalyst is no longer super important, but macOS is.

Why isn't Catalyst super important? 
With all due respect I'm a bit dumbfounded...

**Comment by ghost** [comment]
Is this feature coming anytime soon?

**Comment by lilhinx** [comment]
@jimmycooks9 My rationale for point 2 is that in the years since the announcement of Catalyst, support for Multiplatform SwiftUI apps has increased to the point where Catalyst support is less important to me. I've edited my comment to reflect that.

**Comment by Moriquendi** [comment]
Any updates? :)

**Comment by richardpiazza** [comment]
_Over 4 years since the original question, and almost 8 months since the last comment…_

Has anyone had any success in integrating the XCFramework (or swift package reference) to support a Catalyst app (i.e. iPad running on Mac) or SwiftUI native macOS app?

For those working on this library is there any roadmap information on wether these things will _ever_ be supported? Apple Silicon has opened the door for many of us, but the complete lack of support for using Twilio on the platform essentially kills any movement. The worst part is the not knowing. If support **will not** be added, then we can at least move on to other using other services.

**Comment by richardpiazza** [comment]
Looks like we have our answer... Feel sorry for those who are affected by the companies decisions, but time to start looking at alternatives.

![TwilioEOL](https://github.com/twilio/twilio-video-ios/assets/1744463/d7485b2b-f923-4b6b-be01-a62a712b9dc1)
